Adding the content for "What's this about?" (#1)
citynetwork/ois2023: markdown/what.md

# What's this about?

<!-- Note -->

I work in Cleura in the education team, and what we do is that we run the learning platforms. Our platforms are based on Open edX, a free open source learning management system.

> Open edX platforms have biannual releases like OpenStack, following a naming convention of botanical tree names. So when I mention the names "Maple", "Lilac" I am reffering to specific releases of the Open edX platform.

Reviewer: Consider that you have a multilingual audience, with many non-native English speakers. They might easily confuse "biannual" (twice a year) with "biennial" (once every two years). To avoid potential misunderstandings, you can just say:

```suggestion
Open edX has two releases a year, just like OpenStack, and Open edX releases follow a naming convention of botanical tree names. 
So when I mention the names "Maple", "Lilac", or "Nutmeg", I am referring to specific releases of the Open edX platform.
```

> Starting from the Maple release of Open edX, the community-supported method of deploying the platform using Ansible playbooks was deprecated. Instead, the upstream decided to transition towards a containerized deployment approach.

Reviewer: Active voice is (almost) always better than passive voice, and simple words are better than complicated ones:

```suggestion
Starting from the Maple release of Open edX, upstream deprecated its prior method of deploying the platform using Ansible playbooks.
Instead, they decided to move towards a containerized deployment approach.
```

> To adapt to this change, we made the switch from OpenStack Heat to Kubernetes on OpenStack Magnum. This shift allowed us to embrace containerization for deploying the Open edX platform, leveraging the benefits it provides in terms of scalability, flexibility, and ease of management.

Reviewer: That second sentence: would you actually *say* that in a conversation? I think you would not. Instead, I think you might use a phrasing like:

```suggestion
To adapt to this change, we made the switch from OpenStack Heat to Kubernetes on OpenStack Magnum. 
This shift allowed us to deploy the Open edX platform in containers, and *that* makes things more scalable and flexible for us.
It's also easier for us to manage.
```

Alongside the deployment changes, we also had a transition in our CI-driven deployment approach. We moved from Gitlab CI to Zuul.
 
This is a summary of the architecture before and after, which also discusses some of the issues we ran into along the way.

## Open edX

<!-- Note -->

As mentioned earlier, our platform is built on Open edX. Now let me provide you with a concise overview of what Open edX is and its fundamental components.

> Harvard University and MIT collaborated to establish edX, a non-profit organization with a mission to promote an open-source vision for the future of education. In 2013, edX made its code open source, enabling universities, corporations, NGOs, government organizations, and entrepreneurs to utilize this powerful tool.

Reviewer: These two are not sentences you would actually *say*, so please rewrite them to convey the same information in simpler words.

> The edX platform comprises of three main components. Firstly, the Learning Management System (LMS) serves as the application through which learners access and engage with the course materials. The LMS provides a user-friendly interface for learners to view and interact with the educational content.

Reviewer: It's either "consists of X", or "comprises X". "Comprises of X" is incorrect.

> Secondly, the platform includes Studio, an advanced course creation tool designed for instructors. Studio empowers instructors to develop and design their courses with ease. This sophisticated tool simplifies the process of creating engaging and interactive learning experiences.

Reviewer: Second sentence: you wouldn't say that.

> Lastly, Django Admin panel, the control house for adminstration and management tasks. It facilitates platform-level changes such as setting permissions and managing data, providing administrators with the tools needed to effectively manage the platform.

Reviewer: Second sentence: you wouldn't say that.

> All the components within the edX platform are meticulously designed and offer intuitive interfaces, ensuring that users can navigate and utilize the tools effortlessly.

Reviewer: This sounds like an advertising phrase. This talk is not an advertisement. You can probably just cut this sentence.

## How we used to run Open edX

<!-- Note -->

When we started building our learning platform on Open edX in 2015, we adopted a deployment strategy that aligned with the community's recommended approach.
We opted to deploy the platform on cloud server instances using Ansible playbooks.

> To achieve a cloud-centric and image-driven deployment, we incorporated OpenStack Heat into our workflow and managed our clusters as Heat stacks. Additionally, we effectively managed the entire system using GitLab CI, leveraging its capabilities to orchestrate the deployment process.

Reviewer: You can rewrite the second sentence as just "We used GitLab CI to invoke Heat to deploy the platform." Use simple words, cut the fluff. This isn't LinkedIn. :)

* Application Isolation: Tutor runs Open edX processes inside Docker containers, providing isolation and encapsulation of the application components.

* Command Line Interface (CLI): Tutor offers a CLI for common administrative tasks, making it easier to manage and control the Open edX platform.

* Plugin System: Tutor utilizes a plugin-driven system that allows users to extend or customize Open edX without modifying the core codebase.

Tutor provides comprehensive documentation that offers detailed information on how to use the distribution effectively. If you want to explore further, you can refer to the official Tutor documentation for more insights.

## Our goals

<!-- Note -->

In order to adopt the new Tutor deployment method, our aim was to identify a solution that aligned with the following key goals:

* CI driven Open edX deployment: Our objective was to achieve a fully automated deployment of the Open edX platform to a Magnum managed Kubernetes cluster, similar to our previous setup, through a CI/CD pipeline.

* Configuration and setup changes via Gerrit/Zuul: All configuration and setup changes to Open edX environment are to be done through our Gerrit/Zuul toolchain.

> * High Availability Clusters across Two Regions with Data Replication: We wanted to have a deployment model that utilized high availability clusters distributed over two different regions, while ensuring data replication between them, as we had done in our previous setup. This would provide redundancy and fault tolerance for uninterrupted service.

Reviewer: Data replication and redundancy: yes, high availability: not really, fault tolerance: no. Yes, Kubernetes gives us HA _within_ a region, but our HA cluster doesn't span two regions.

> * Stateful Configuration Management: We sought to maintain the ability to manage our Open edX platform through a stateful configuration engine. In our previous setup, we utilized Heat stacks, and now we wanted to leverage Kubernetes deployments as the orchestration tool.

Reviewer: Ditch the verbs "utilize" and "leverage". "Use" is a fine replacement for both.

# What's this about?

<!-- Note -->

I work in Cleura in the education team, and what we do is that we run the learning platforms. Our platforms are based on Open edX, a free open source learning management system.

> Open edX has two releases a year, just like OpenStack, and Open edX releases follow a naming convention of botanical tree names. So when I mention the names "Maple", "Lilac", or "Nutmeg", I am reffering to specific releases of the Open edX platform.

Reviewer: Maybe also mention Olive here, since it is the current release.

Open edX is an open-source learning management system (LMS) an online course platform that allows individuals and organizations to create, deliver, and manage online educational content. 	

The edX platform consists of three main components. Firstly, the Learning Management System (LMS) serves as the application through which learners access and engage with the course materials. The LMS provides a user-friendly interface for learners to view and interact with the educational content.

Secondly, the platform includes Studio, an advanced course creation tool designed for instructors. It is the content management system for creating courses and course libraries in the Open edX platform.

Lastly, Django Admin panel, which allows the administrators to handle tasks such as magnaging data and setting permissions.

## How we used to run Open edX

<!-- Note -->

When we started building our learning platform on Open edX in 2015, we adopted a deployment strategy that aligned with the community's recommended approach.
We opted to deploy the platform on cloud server instances using Ansible playbooks.

> To achieve a cloud-centric and image-driven deployment, we incorporated OpenStack Heat into our workflow and managed our clusters as Heat stacks. We used Gitlab CI toinvoke Heat to deploy the platform.

Reviewer: Typo:

```suggestion
To achieve a cloud-centric and image-driven deployment, we incorporated OpenStack Heat into our workflow and managed our clusters as Heat stacks. We used Gitlab CI to invoke Heat to deploy the platform.
```
